graphistry/feature_utils.py

```diff
@@ -2495,13 +2649,21 @@ def featurize(
                 default True.
         :return: graphistry instance with new attributes set by the featurization process.
         """
-        assert_imported()
+        try:
+            feature_engine = resolve_feature_engine(feature_engine)
+        except:
+            feature_engine = resolve_feature_engine(engine)
```

Comment:
i'm confused why this is necessary?

also, to keep mypy smoother, may help to do `resolved_feature_engine : FeatureEngineConcrete = ...` so that the `featurize(feature_engine: T1)` and `resolved_feature_engine : T2` typings do not conflict